lintcheck/src/main.rs

```diff
@@ -374,25 +373,17 @@ impl Crate {
             );
         }
 
-        let cargo_clippy_path = fs::canonicalize(cargo_clippy_path).unwrap();
-
         let shared_target_dir = clippy_project_root().join("target/lintcheck/shared_target_dir");
 
-        let mut cargo_clippy_args = if config.fix {
-            vec!["--quiet", "--fix", "--"]
-        } else {
-            vec!["--quiet", "--message-format=json", "--"]
-        };
-
         let cargo_home = env!("CARGO_HOME");
 
         // `src/lib.rs` -> `target/lintcheck/sources/crate-1.2.3/src/lib.rs`
         let remap_relative = format!("={}", self.path.display());
         // Fallback for other sources, `~/.cargo/...` -> `$CARGO_HOME/...`
         let remap_cargo_home = format!("{cargo_home}=$CARGO_HOME");
-        // `~/.cargo/registry/src/github.com-1ecc6299db9ec823/crate-2.3.4/src/lib.rs`
+        // `~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/crate-2.3.4/src/lib.rs`
         //     -> `crate-2.3.4/src/lib.rs`
-        let remap_crates_io = format!("{cargo_home}/registry/src/github.com-1ecc6299db9ec823/=");
+        let remap_crates_io = format!("{cargo_home}/registry/src/index.crates.io-6f17d22bba15001f/=");
```

Comment:
This change is because the crates.io sparse registry uses a different dir to the old git based one which was the default when I wrote the initial implementation

Reply:
How is the hash of this dir built? Doesn't this hash update when cargo/crates.io updates something? :thinking:

Reply:
It appears to be derived from the full URL of the registry https://users.rust-lang.org/t/where-does-6f17d22bba15001f-come-from-in-index-crates-io-6f17d22bba15001f/110225/3

Reply:
Thanks for the explanation!